tests/unit/helpers.js

```diff
@@ -274,6 +274,70 @@ export const fakeAddonInfo = {
   privacy_policy: ' some privacy policy text',
 };
 
+export const fakePrimaryHeroShelfExternal = Object.freeze({
+  id: 1,
+  guid: 'some-guid',
+  homepage: 'https://mozilla.org',
+  name: 'some external name',
+  type: ADDON_TYPE_EXTENSION,
+});
+
+export const createPrimaryHeroShelf = ({
+  addon = fakeAddon,
+  description = 'Primary shelf description',
+  external = undefined,
+  featuredImage = 'https://addons-dev-cdn.allizom.org/static/img/hero/featured/teamaddons.jpg',
+  gradient = { start: '000000', end: 'FFFFFF' },
+} = {}) => {
+  return {
+    addon,
+    description,
+    external,
+    featured_image: featuredImage,
+    gradient,
+  };
+};
+
+export const createSecondaryHeroShelf = ({
+  cta = { url: 'https://mozilla.org', text: 'Some cta text' },
+  description = 'Secondary shelf description',
+  headline = 'Secondary shelf headline',
+  modules = [
+    {
+      icon: 'icon1',
+      description: 'module 1 description',
+      cta: { url: 'https://mozilla.org/1', text: 'module 1 cta text' },
+    },
+    {
+      icon: 'icon2',
+      description: 'module 2 description',
+      cta: { url: 'https://mozilla.org/2', text: 'module 2 cta text' },
+    },
+    {
+      icon: 'icon3',
+      description: 'module 3 description',
+      cta: { url: 'https://mozilla.org/3', text: 'module 3 cta text' },
+    },
+  ],
+} = {}) => {
+  return {
+    cta,
+    description,
+    headline,
+    modules,
+  };
+};
```

Comment:
I forgot how Flow works. Can we use the defined Flow types in the reducer here so that it validates these fake objects?

Reply:
We don't use Flow for tests in addons-frontend, so this file isn't currently covered by Flow. If I add `/* @flow */` at the top of this file I get *94* Flow errors. While I agree that this might be a nice thing to have, I don't think it's worth the effort it would require for this patch. I can file a follow-up to try to add Flow coverage to this file, but, considering that we don't have any Flow coverage for any of our tests, I'm not sure how high a priority it would be to address the issue for this file.

Reply:
ah, well, ok.. Forget about this idea 😋